index.js

```diff
@@ -1,88 +1,117 @@
+// // => TypeError
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
-
-// console.log(drinkWater(12))
+console.log("%cQuestion 3", "color: red") 
+function drinkWater(currentThirstLevel){
+  console.log("Man I sure am thirsty")
+  currentThirstLevel -= 1
+  console.log("Ahh that hits the spot")
+  return currentThirstLevel
+}
+// def drink_water(current_thirst_level)
+//   puts "Man I sure am thirsty"
+//   current_thirst_level -= 1
+//   puts "Ahh that hits the spot"
+//   current_thirst_level
+// end
+console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
-
-// console.log(sameSameButDifferent(5, 5)) 
+console.log("%cQuestion 4", "color: red") 
+function sameSameButDifferent (num, maybeNum){
+  if (num === maybeNum){
+    return "same same"
+  } else if (num == maybeNum){
+    return "same same (but different)"
+  } else {
+    return "different"
+  }
+}
+console.log(sameSameButDifferent(5, 5)) 
 // // => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
+console.log(sameSameButDifferent(123, "123")) 
 // // => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
+console.log(sameSameButDifferent(5, 7)) 
 // // => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
+console.log(sameSameButDifferent(123, "122")) 
 // // => "different"
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
+console.log("%cQuestion 5", "color: red")
+
+function updateGrade (student, grade){
+  student.grade = grade
+}
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+console.log(student1)
 // // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+console.log("%c----------", "color: red")
 
 
 
 // ***** Question 6 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
+console.log("%cQuestion 6", "color: red")
+function printNameAndPhones(users){
+  users.forEach(element => console.log(element.name))
+  users.forEach(element => console.log(element.phones.cell))
+  users.forEach(element => console.log(element.phones.office))
+}
```

Comment:
Nice job using arrow functions! I see you're iterating over the array 3 times. The functionality required in this function to be done in one iteration, which is better for performance.

- Here's how it would look using an arrow function as the callback to `forEach` in one iteration:
```javascript
  users.forEach(user => {
    console.log(user.name);
    console.log(`Cell: ${user.phones.cell}`);
    console.log(`Office: ${user.phones.office}`);
  })
```